src/passes/inlineAssemblyTransformer.ts

```diff
@@ -217,22 +216,22 @@ export class InlineAssemblyTransformer extends ASTMapper {
     throw new WillNotSupportError(errorMsg, undefined, false);
   }
 
-  transform(node: InlineAssembly, ast: AST): void {
+  transform({ node, ast }: { node: InlineAssembly; ast: AST; }): void {
     assert(node.context !== undefined, `Assembly root context not found.`);
-    const transformer = new YulTransformer(ast, node.externalReferences, node.context);
+    const transformer = new YulTransformer({ ast, externalReferences: node.externalReferences, context: node.context });
 
-    const statements = this.yul(node).statements.map((yul: YulNode) => {
-      const astNode = transformer.toSolidityStatement(yul);
+    const statements = this.yul({ node }).statements.map((yul: YulNode) => {
+      const astNode = transformer.toSolidityStatement({ node: yul });
       ast.setContextRecursive(astNode);
       return astNode;
     });
-    const block: Block = new UncheckedBlock(ast.reserveId(), this.yul(node).src, statements);
+    const block: Block = new UncheckedBlock(ast.reserveId(), this.yul({ node }).src, statements);
     ast.replaceNode(node, block);
   }
 
   visitInlineAssembly(node: InlineAssembly, ast: AST): void {
-    this.verify(node, ast);
-    this.transform(node, ast);
+    this.verify({ node, ast });
```

Comment:
Why are wrapping the arguments in an object?
What lint warning is it solving, is there a way we can modify the lint rule instead